File: 14697.cpp
#include <iostream>
using namespace std;

int main() {
	int A, B, C, N;
	cin >> A >> B >> C >> N;

	int room1, room2, room3;
	for (int i = 0; i <= N; i++) {
		room1 = A * i;
		for (int j = 0; j <= N; j++) {
			room2 = B * j;
			for (int k = 0; k <= N; k++) {
				room3 = C * k;

				if (room1 + room2 + room3 == N) {
					cout << 1 << endl;
					return 0;
				}
			}
		}
	}

	cout << 0 << endl;
}

File: 14697.py
A, B, C, N = map(int, input().split())

for i in range(N + 1):
    room1 = A * i
    for j in range(N + 1):
        room2 = B * j
        for k in range(N + 1):
            room3 = C * k

            if room1 + room2 + room3 == N:
                print(1)
                exit()

print(0)
